framegear/components/Frame/Frame.tsx

```diff
@@ -140,7 +152,7 @@ function FrameButton({
       openModal(onConfirm);
     }
     // TODO: implement other actions (mint, etc.)
-  }, [button, setResults]);
+  }, [button?.action, button?.index, button?.target, inputText, openModal, setResults]);
```

Comment:
Had a warning here because `openModal` wasn't included as dep. The other formatting was done automatically by IDE.